test/unit/helpers/mocks/Util.js

```diff
@@ -313,6 +314,12 @@ function (Okta, Q, Duo) {
     return responseCopy;
   };
 
+  fn.transformMockXHR = function (xhr) {
+    // The xhr needs to be transformed without modifying the existing response
+    var xhrClone = JSON.parse(JSON.stringify(xhr));
+    return LoginUtil.transformErrorXHR(xhrClone);
+  }
```

Comment:
In order to validate that our `xhr` objects contain the expected information, we need to do two things:
1. Run it through the [`transformErrorXHR`](https://github.com/okta/okta-signin-widget/blob/master/src/widget/OktaSignIn.js#L215) method, similar to what the Widget does by default. This maps the request object correctly.
2. Clone the `xhr` by using the `JSON.parse` and `JSON.stringify` hack to avoid overriding the original value. Without this, the API response we're taking from `helpers/xhr/*` will be modified and reused throughout our tests.

I was unable to find a better way to do this, so I'm open to suggestions.

Reply:
I usually prefer to assert with plain object (if it's not too verbose at this case)
Pros is if any mistakes / changes in `transformErrorXHR`, test would be able to catch. (an extreme example is if `transformErrorXHR` return `{}`, test would still pass but apparently that's not the right implementation)
Const is of course is very verbose to type of the actual `xhr` value in test.